plugins/inputs/execd/execd_unix.go

```diff
@@ -0,0 +1,33 @@
+// +build !windows
+
+package execd
+
+import (
+	"fmt"
+	"io"
+	"syscall"
+
+	"github.com/influxdata/telegraf"
+)
+
+func (e *Execd) Gather(acc telegraf.Accumulator) error {
+	if e.cmd == nil || e.cmd.Process == nil {
+		return nil
```

Comment:
Is this a good place to restart `cmdRun`?

Reply:
It is surly possible.

It would mean that the process can be dead up to the configured `interval`. If the execd program is collecting metrics by itself or listens for events this can be quite long.

Reply:
It seems like this is a reasonable trade-off to always restarting (even on `Stop`) I'm curious to hear what @danielnelson thinks

Reply:
Restarting on `Stop` has been fixed. Processed will not be restarted if the plugin has been stopped (3d603c4).